### PR TITLE
MDL-71696 core_question: changes for question status and view

### PR DESCRIPTION
**Pull request by safatshahin**
PR for question status and question base view.

### CODE REVIEW

File: question/bank/deletequestion/delete.php [hunk 1]
```diff
@@ -77,7 +77,7 @@
                 $questionid = (int)$questionid;
                 question_require_capability_on($questionid, 'edit');
                 if (questions_in_use(array($questionid))) {
-                    $DB->set_field('question', 'hidden', 1, array('id' => $questionid));
+                    $DB->set_field('question_versions', 'status', 1, ['questionid' => $questionid]);
```
Comment (guillogo) [inline question/bank/deletequestion/delete.php:80]: Maybe we need something similar in line 61 https://github.com/catalyst/moodle-MDL-70329/pull/259/files#R61

File: question/bank/viewcreator/templates/version_info.mustache [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        "addquestiondata": [
+          {
+                "version": "version html",
+                "extrahtml": "HTML for elements",
+          }
+        ]
+    }
+}}
+<div>
```
Comment (guillogo) [inline question/bank/viewcreator/templates/version_info.mustache:30]: can you a add an id to identify the selector if some one wants to change the style in their own themes.

File: question/classes/local/bank/view.php [hunk 3]
```diff
@@ -547,7 +549,12 @@ protected function build_query(): void {
         }
 
         // Build the where clause.
-        $tests = ['q.parent = 0'];
+        $latestversion = 'qv.version  = (SELECT MAX(v.version)
+                                        FROM {question_versions} v
```
Comment (guillogo) [inline question/classes/local/bank/view.php:553]: can you align this to the select please

File: question/classes/local/bank/view.php [hunk 4]
```diff
@@ -1141,7 +1146,7 @@ protected function print_table_headers(): void {
      */
     protected function get_row_classes($question, $rowcount): array {
         $classes = [];
-        if ($question->hidden) {
+        if ((int)$question->status == 1) {
```
Comment (guillogo) [inline question/classes/local/bank/view.php:1149]: as you are adding a 'int cast' we can compare using `===`

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @safatshahin 

Thank you for this, just minor comments in the code and 1 more thing:

1. Can you check the value in this object is getting the right status: https://github.com/catalyst/moodle-MDL-70329/blob/master_MDL-71696-versioning-question-status/question/bank/deletequestion/classes/delete_action_column.php#L87

Also, there is an error when we try to save a qtype_multianswer but this will be fixed on that branch, the children question will need the parent status.

Thank you.

**Comment by safatshahin** [comment]
Hi @guillogo 
Thank you for your review of this. I have made the changes. Please have a look now.

**Review by guillogo — APPROVED** [review]
Thank you @safatshahin is good now, approved.

**Pull request merged**